src/main/java/stream/flarebot/flarebot/Events.java

```diff
@@ -96,13 +96,15 @@ public void onMessageReactionAdd(MessageReactionAddEvent event) {
                     button.onClick(ButtonUtil.getButtonGroup(event.getMessageId()).getOwner(), event.getUser());
                     String emote = event.getReactionEmote() != null ? event.getReactionEmote().getName() + "(" + event.getReactionEmote().getId() + ")" : button.getUnicode();
                     Metrics.buttonsPressed.labels(emote).inc();
-                    event.getChannel().getMessageById(event.getMessageId()).queue(message -> {
-                        for (MessageReaction reaction : message.getReactions()) {
-                            if (reaction.getReactionEmote().equals(event.getReactionEmote())) {
-                                reaction.removeReaction(event.getUser()).queue();
+                    if(event.getGuild().getSelfMember().hasPermission(Permission.MESSAGE_MANAGE)) {
```

Comment:
Formatting. But you may as well just return here. Think it will be a bit more readable as it's just becoming really intended here :D